### PR TITLE
46 send queue listener implementation

### PR DESCRIPTION
**Pull request by sahul343**
Fixes convict-git/KoraKaagaz#46

### CODE REVIEW

File: src/networking/SendQueueListener.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+     * send over the network.
+     */
+    private ConcurrentBlockingQueue<OutgoingPacket> SendQueue;
+
+
+    /**
+     * Constructor for this sendQueueListener class.
+     * @param SendQueue, which require the sendqeue as parameter.
+     */
+    public SendQueueListener(ConcurrentBlockingQueue<OutgoingPacket> SendQueue){
```
Comment (sravand123) [inline src/networking/SendQueueListener.java]: use IQueue in the parameters of the constructors

File: src/networking/SendQueueListener.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+     * @return boolean, if it's valid port then returns true, else return false
+     */
+    private boolean isValidPort(String port){
+        String intPattern = "[0-9]+";
+        if(Pattern.matches(intPattern, port)){
+            int value = Integer.parseInt(port);
+            //it will return true if port is valid else false.
+            return (value > 0) && (value <= 65535);
+        }
+
```
Comment (sravand123) [inline src/networking/SendQueueListener.java:102]: ``` return false```

**Review by sravand123 — CHANGES_REQUESTED** [review]
use Logger Wherever necessary

File: src/networking/SendQueueListener.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * The method will check whether ip:port address is valid or not.
+     * @param destination, is ip, port address as string
+     * @return boolean, return true if destination address is valid, false if it's not valid.
+     */
+    private boolean isValidAddress(String destination){
+        String[] dest = splitAddress(destination);
+        return ( dest.length == 2 && isValidIpaddress(dest[1]) && isValidPort(dest[0]) );
```
Comment (sravand123) [inline src/networking/SendQueueListener.java]: The format of destination is ip:port,so this statement should be
```isValidIpaddress(dest[0]) && isValidPort(dest[1])```

File: src/networking/SendQueueListener.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    /**
+     * This method will do the work of taking the data from the queue 
+     * and send it over the network.
+     */
+    public void run(){
+        
+        /**
+         * run the while loop as long as the application is running.
+         */
+        while(LanCommunicator.isRunning){
```
Comment (sravand123) [inline src/networking/SendQueueListener.java]: Use ```getStatus``` from LanCommunicator to check the status

**Review by sravand123 — CHANGES_REQUESTED** [review]

**Comment by sahul343** [comment]
> use Logger Wherever necessary

Thanks. used logger whenever it is required. Have a look

**Review by sravand123 — APPROVED** [review]
Everything is working....Excellent Work sahul...

**Comment by sravand123** [comment]
[SendQueueListener](https://trello.com/c/T6ZuBaMd/5-sendqueuelistener)

**Pull request merged**